backend/document/domain/document_generator.py

```diff
@@ -611,23 +729,86 @@ def main(document_request: model.DocumentRequest) -> tuple[str, str]:
-            output_filename,
-            document_request_key_,
-            document_request,
+        content = assemble_content(
+            document_request_key_, document_request, book_content_units
+        )
+        write_html_content_to_file(
+            content,
+            html_filepath_,
+        )
+        write_html_cover_to_file(
+            cover_filepath_,
             book_content_units,
             unfound_resource_lookup_dtos,
             unloaded_resource_lookup_dtos,
         )
 
+    # Immediately return pre-built PDF if the document has previously been
+    # generated and is fresh enough. In that case, front run all requests to
+    # the cloud including the more low level resource asset caching
+    # mechanism for comparatively immediate return of PDF.
+    if document_request.generate_pdf and file_utils.asset_file_needs_update(
+        pdf_filepath_
+    ):
+        logger.info("Generating PDF %s...", pdf_filepath_)
+        convert_html_to_pdf(
+            html_filepath_,
+            cover_filepath_,
+            pdf_filepath_,
+        )
+        copy_pdf_to_docker_output_dir(pdf_filepath_)
+
+    if document_request.generate_epub and file_utils.asset_file_needs_update(
+        epub_filepath_
+    ):
+        convert_html_to_epub(
+            html_filepath_,
+            cover_filepath_,
+            epub_filepath_,
+        )
+        copy_epub_to_docker_output_dir(epub_filepath_)
+
+    if document_request.generate_docx and file_utils.asset_file_needs_update(
+        docx_filepath_
+    ):
+
+        convert_html_to_docx(
+            html_filepath_,
+            cover_filepath_,
+            docx_filepath_,
+        )
+        copy_docx_to_docker_output_dir(docx_filepath_)
+
     if should_send_email(document_request.email_address):
-        send_email_with_pdf_attachment(
-            document_request.email_address, output_filename, document_request_key_
+        if document_request.generate_pdf:
+            attachments = [
+                model.Attachment(
+                    filepath=pdf_filepath_, mime_type=("application", "octet-stream")
+                )
+            ]
+        if document_request.generate_epub:
+            attachments.append(
+                model.Attachment(
+                    filepath=epub_filepath_, mime_type=("application", "octet-stream")
+                )
+            )
+        if document_request.generate_docx:
+            attachments.append(
+                model.Attachment(
+                    filepath=docx_filepath_, mime_type=("application", "octet-stream")
+                )
+            )
+
+        send_email_with_attachment(
+            document_request.email_address,
+            attachments,
+            document_request_key_,
         )
-    return document_request_key_, output_filename
+    return document_request_key_
 
 
+# FIXME Turns out tn-wa and tn really are different, but only one should
+# be used. Need to find out which is preferred.
```

Comment:
tn-wa, which I believe only exists for English, is always preferred over tn.

Reply:
Thanks for answering this question! I will address this in an upcoming PR then.